binder/helpers.py

```diff
@@ -92,17 +92,16 @@ def delete_record(dns_server, rr_list, key_name):
         keyring = transfer_key.create_keyring()
         algorithm = transfer_key.algorithm
 
-    delete_response = []
-    for current_rr in rr_list:
-        record_list = current_rr.split(".")
-        record = record_list[0]
-        domain = ".".join(record_list[1:])
-        dns_update = dns.update.Update(domain, keyring=keyring, keyalgorithm=algorithm)
-        dns_update.delete(record)
-        output = send_dns_update(dns_update, dns_server, key_name)
-
-        delete_response.append({ "description" : "Delete Record: %s" % current_rr,
-                                 "output" : output })
+    dns_update = dns.update.Update(zone_name, keyring=keyring, keyalgorithm=algorithm)
+    records = []
+    for resource_record in rr_list:
+        dns_update.delete(resource_record)
+        records.append("%s.%s" % (resource_record, zone_name))
+
+    output = send_dns_update(dns_update, dns_server, key_name)
+    delete_response = [{"description": "Deleted Records:<br/>%s" %
```

Comment:
I refuse to put any sort of formatting of the data in the Python itself. Leave the output formatting for the templates.